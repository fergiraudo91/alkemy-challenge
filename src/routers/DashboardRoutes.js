import React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'
import { HeroScreen } from '../components/heroes/HeroScreen'
import { HomeScreen } from '../pages/HomeScreen'
import { Navbar } from '../components/ui/Navbar';
import { SearchScreen } from '../pages/SearchScreen';

export const DashboardRoutes = () => {
    
    return (
        <>
            <Navbar />
                <Switch>
                    <Route exact path="/home" component={ HomeScreen } />
                    <Route exact path="/hero/:heroeId" component={ HeroScreen } />
                    <Route exact path="/search" component= { SearchScreen } />
                    <Redirect to="/home" />
                </Switch>
        </>
    )
}
