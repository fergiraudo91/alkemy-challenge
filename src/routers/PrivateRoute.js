import React from 'react';
import { Redirect, Route } from 'react-router-dom';
import PropTypes from 'prop-types';

export const PrivateRoute = ({
    isAuthenticated,
    component: Component,
    ...rest
}) => {
    localStorage.setItem('lastPath', rest.location.pathname);
    const token = localStorage.getItem('token');
    return (
       <Route
       component = {(props) => (
           (isAuthenticated && token)
           ? <Component {...props}/>
           : <Redirect to="/login" />
       )}
       {...rest}
       />
    )
}

PrivateRoute.propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    component: PropTypes.func.isRequired
}
