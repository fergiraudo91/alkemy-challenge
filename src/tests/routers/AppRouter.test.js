import React from 'react';
import { mount } from "enzyme";
import { AppRouter } from "../../routers/AppRouter";
import { AuthContext } from '../../auth/AuthContext';

describe('Testing <AppRouter />', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }

    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <AppRouter />
        </AuthContext.Provider>
    )

    test('should show the login if it is not authenticated', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.login-screen').exists()).toBe(true);
    });


});
