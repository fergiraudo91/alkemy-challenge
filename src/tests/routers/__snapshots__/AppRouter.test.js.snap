// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <AppRouter /> should show the login if it is not authenticated 1`] = `
<div
  className="main-login-screen"
>
  <div
    className="login-screen"
  >
    <img
      alt="batman logo"
      src="assets/img/batman-ico.png"
    />
    <form
      className="login-form"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="field"
          htmlFor="exampleInputEmail1"
        >
          User
        </label>
        <input
          className="form-control"
          id="exampleInputEmail1"
          name="email"
          onChange={[Function]}
          required={true}
          type="mail"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="field"
          htmlFor="exampleInputPassword1"
        >
          Password
        </label>
        <input
          className="form-control"
          id="exampleInputPassword1"
          name="password"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
      </div>
      <button
        className="btn btn-light"
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;
